### PR TITLE
Format date attribute headers in execution

### PR DESCRIPTION
**Pull request by scavnickyj**
When backend sends granularity and format information for a date attribute header in executionResponse, we format the date header accordingly. These new formatted names are used in visualisations instead of the original names. However, they should not be used in drilling as this may cause inconsistencies.

- extend set of supported locales
- extend default date formatter to accept custom formatting patterns
- update default date formatting patterns according to patterns from BE
- update date parsing patterns
- extend IResultAttributeHeaderItem to hold formattedName
- use formattedName in visualisations
- ensure drilling in pivot table does not return formattedName

JIRA: TNT-1106

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by scavnickyj** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1878/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10309/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1878/ : UNSTABLE in 28m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10311/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12300/ : SUCCESS in 20m 22s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10313/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12301/ : SUCCESS in 23m 53s

**Comment by scavnickyj** [comment]
extended test

File: libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/dateValueParser.ts [hunk 1]
```diff
@@ -10,9 +10,9 @@ const granularityParseValueTransformations: {
     [granularity in DateAttributeGranularity]?: ValueTransform;
 } = {
     "GDC.time.day_in_week": (value) => {
-        // server returns 0 = Sunday, 6 = Saturday
-        // date-fns expects 1 = Monday, 7 = Sunday
-        return value === "0" ? "7" : value;
+        // server returns 00 = Sunday, 06 = Saturday
+        // date-fns expects 1 = Sunday, 7 = Saturday
+        return `${parseInt(value) + 1}`;
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/dateValueParser.ts:15]: Is this true?
See https://date-fns.org/v2.29.3/docs/setDay
It seems it uses the same logic as the server, or am I missing something?

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1881/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10317/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1881/ : SUCCESS in 24m 45s

File: libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/dateValueParser.ts [hunk 2]
```diff
@@ -10,9 +10,9 @@ const granularityParseValueTransformations: {
...
     },
 };
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/dateValueParser.ts:18]: Maybe some link to https://date-fns.org/docs/format or https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table would be useful here (so it's clear, which standard it uses)

File: libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/defaultDateFormatter.ts [hunk 3]
```diff
@@ -1,32 +1,40 @@
...
 import { UnexpectedError } from "@gooddata/sdk-backend-spi";
-import { DateFormatter } from "./types";
+import identity from "lodash/identity";
 
-const granularityFormatPatterns: {
+const defaultLocaleCode = "en-US";
+
+/**
+ * Default date format patterns for all granularities.
+ * They should be identical to the ones that backend sends in case of en-US locale.
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/dateFormatting/defaultDateFormatter.ts:12]: Again, some link to https://date-fns.org/docs/format or https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table would be useful here (so it's clear, which standard it uses)

File: libs/sdk-ui-pivot/src/impl/drilling/drilledRowFactory.ts [hunk 4]
```diff
@@ -67,7 +67,18 @@ export function createDrilledRow(row: IGridRow, tableDescriptor: TableDescriptor
                 result.push(createDrilledSliceDetail(null, null, null));
                 break;
             default:
-                result.push(createDrilledSliceDetail(id, drillItemUri, row[col.id]));
+                if (hasMappingHeaderFormattedName(mappingHeader)) {
+                    // we want to use the original name insted of the formatted name in drilling
```
Comment (kandl) [inline libs/sdk-ui-pivot/src/impl/drilling/drilledRowFactory.ts:71]: typo: instead

**Review by kandl — APPROVED** [review]
LGTM

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Pull request merged**